### ISSUE TITLE
How does composables know about keyboard

### ISSUE THREAD

**Issue by shalva97** [kiwicom/orbit-compose#148]
in the Text Fields controls if I focus a field and keyboard shows up then TextFields also scroll and appear above primary button
![Screenshot 2022-05-17 at 22 47 40](https://user-images.githubusercontent.com/22417494/168887784-20c2dd83-8294-480f-8e1f-6d98188df37f.png)

How does that work? on my app, with your components, this is the results
![Screenshot 2022-05-17 at 22 49 14](https://user-images.githubusercontent.com/22417494/168888101-ed6c4df6-77da-4aef-a238-c8dafedc28f5.png)
TextField is below the keyboard with textSelectHandle on top of it :D

Catalog app has `android:windowSoftInputMode="adjustResize"`, but adding it on my app has no effect.

My layout is like this:

```kotlin
@Preview(showSystemUi = true)
@Composable
fun LoginScreen() {

    val viewModel: LoginScreenViewModel = hiltViewModel()

    var server by remember { mutableStateOf("192.168.") }

    val discoveredServer = viewModel.foundServers.collectAsState(initial = emptyList())

    Column(modifier = Modifier
        .systemBarsPadding()
        .fillMaxSize()
        .padding(10.dp),
        verticalArrangement = Arrangement.Bottom) {

        LazyColumn {
            items(discoveredServer.value.size) { index ->
                Card(onClick = { viewModel.connectToServer(discoveredServer.value[index]) }) {
                    Text(text = discoveredServer.value[index].address ?: "")
                }
            }
        }

        TextField(value = server,
            onValueChange = { server = it },
            label = { Text(text = "Server") },
            leadingIcon = { Icon(Icons.Trip, contentDescription = null) },
            modifier = Modifier.fillMaxWidth(),
            maxLines = 1)

        Button(onClick = { /*TODO*/ }, modifier = Modifier.align(Alignment.End), enabled = false) {
            Text(text = "Next", modifier = Modifier)
        }
    }
}
```

which is included inside `OrbitTheme` -> `Scaffold` -> Jetpack Navigation...

**Comment by hrach** [comment]
So are you using Orbit's TextField? If yes, there is the functionality to bring it into focus (see its source code). 

We also utilize some measuring and Locals to add additional padding to bring the button above Scaffold's CTA.

**Comment by shalva97** [comment]
Yes, TextField is from Orbit. One thing I noticed is that, in the sample app, there is `.verticalScroll(rememberScrollState())` modifier for Box. This is what I also dont have and without this the sample app also works like mine...

So is the verticalScroll needed including Scaffold? 

Also could you please explain the expected layout for Orbit design? I was thinking about removing Scaffold entirely, but it looks like TextField sort of depends on Scaffold, right?

**Comment by hrach** [comment]
> So is the verticalScroll needed including Scaffold?

I'm not sure I follow, but yes, scrollable parent is needed if you want to bring TextFiled above the keyboard.

> Also could you please explain the expected layout for Orbit design? I was thinking about removing Scaffold entirely, but it looks like TextField sort of depends on Scaffold, right?

There is no requirement to use Scaffold to design a screen. You can be good with Column, Surface, ... Scaffold is a helper for bottom action layouting and topappbar layouting. At least for now.
